src/cider/nrepl/middleware/util/storage.clj

```diff
@@ -0,0 +1,16 @@
+(ns cider.nrepl.middleware.util.storage)
+
+;; TODO: How should we expire the cache?
+;; Currently effectively single use storage for use with non-sync exceptions,
+;; but since this can likely be used for any type of data,
+;; perhaps we should have some formal autopurge functionality so that this
+;; doesn't grow unbounded?
+(def ^:dynamic *storage* (atom {}))
```

Comment:
I'm also not sure about the structure of the cache - if this is going to be used for other things it'd be hard to keep track of what is what with the current implementation.